classes/local/skel/lib_php_file.php

```diff
@@ -0,0 +1,61 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Provides tool_pluginskel\local\skel\lib_php_file class.
+ *
+ * @package     tool_pluginskel
+ * @subpackage  skel
+ * @copyright   2016 Alexandru Elisei <alexandru.elisei@gmail.com>, David Mudrák <david@moodle.com>
+ * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace tool_pluginskel\local\skel;
+
+use tool_pluginskel\local\util\exception;
+
+defined('MOODLE_INTERNAL') || die();
+
+/**
+ * Class representing the plugin's lib.php file.
+ *
+ * @copyright   2016 Alexandru Elisei <alexandru.elisei@gmail.com>
+ * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class lib_php_file extends php_internal_file {
+
+    /**
+     * Set the data to be eventually rendered.
+     *
+     * @param array $data
+     */
+    public function set_data(array $data) {
+
+        parent::set_data($data);
+
+        // The plugin must advertise the fact it supports making a backup.
```

Comment:
No, any file skeleton can only be added once because of these [lines](https://github.com/mudrd8mz/moodle-tool_pluginskel/blob/master/classes/local/util/manager.php#L178):
```php
if (isset($this->files[$filename])) {
    throw new exception('The file has already been initialised: '.$filename);
}
```

That's when the recipe gets modified to have options like $data['has_supported_features'] = true, which in turn will lead to include the relevant function from the template.